### PR TITLE
fix: refactor all stacking issues (z-index)

### LINKED ISSUE #1038: Tooltip z-index issue on design system section.
### Does a discussion already exist on the topic you are interested in, or on a similar topic?

- [X] I verified and it doesn't exist

### What

There is a z-index issue with tooltips overlapped by design system navigation sidebar. See image. 

issue

### Why

It is a bug. 

### Context

Maybe a regression inserted by https://github.com/italia/designers.italia.it/commit/859d90506fcf934bd45b704ea9ec18533aa80176 @bfabio @Lorenzo-Pappalardo ?

### More

_No response_

### PR DESCRIPTION
**Pull request by Fupete**
Trying to leave just the minimum amount of hacks as possibile. 

To be resolved soon on BSI, see https://github.com/italia/bootstrap-italia/issues/1009

Fix #1038 .

### CODE REVIEW

File: src/components/nav-sidebar/nav-sidebar.scss [hunk 1]
```diff
@@ -86,7 +86,7 @@
 //Mobile & Tablet portrait only -------------------------------------------------------------------------
 @include media-breakpoint-down(lg) {
   .menu-column {
-    z-index: 10010 !important;
+    z-index: 10001 !important; // XXX $zindex-modal-backdrop
```
Comment (Fupete) [inline src/components/nav-sidebar/nav-sidebar.scss:89]:
```suggestion
    z-index: 10001 !important; // XXX usually $zindex-modal-backdrop but not >9999 to be above fixed sticky header
```

File: src/components/nav-sidebar/nav-sidebar.scss [hunk 2]
```diff
@@ -96,13 +96,14 @@
...
     overflow-y: auto;
+    z-index: 0 !important; // content level also with bs-is-sticky
   }
 
   .nav-sidebar {
     .it-navscroll-wrapper {
       background-color: transparent !important;
       .menu-wrapper {
-        z-index: 0;
+        /* z-index: 0; */
```
Comment (Fupete) [inline src/components/nav-sidebar/nav-sidebar.scss:106]:
```suggestion
```

File: src/components/highlight/highlight.scss [hunk 3]
```diff
@@ -22,7 +22,7 @@
 		.icon,
     .overlay-image {
 			position: absolute;
-			z-index: 1;
+			/*z-index: 1;*/
```
Comment (Fupete) [inline src/components/highlight/highlight.scss:25]:
```suggestion
```

File: src/components/header-menu-link/header-menu-link.scss [hunk 4]
```diff
@@ -110,7 +110,7 @@ $navigation-link-active-bar-size: 3px;
         opacity: 0;
         display: block;
         transition: opacity 0.15s cubic-bezier(0.1, 0.57, 0.36, 0.99);
-        z-index: 2;
+        /* z-index: 2; */
```
Comment (Fupete) [inline src/components/header-menu-link/header-menu-link.scss:113]:
```suggestion
```

**Comment by Fupete** [comment]
@bfabio we have to remove from this PR the commit of `piano-attivita.yaml`. It is here by mistake. Just remove it, it is already in another PR.

**Comment by Fupete** [comment]
@bfabio I just commented above some proposals just to remind to remove commented SCSS lines before merging.

**Comment by Fupete** [comment]
@bfabio I found out that this resolve almost everything we need. Already said that BSI must resolve all z-index by default asap... 

However I found a little bug in this PR where nav-sidebar is present on mobile... If you open the menu the sidebar is there on the bottom... but not really a bug, maybe a feature? 🤔 
[immagine](https://github.com/italia/designers.italia.it/assets/6736103/2725fec8-8a74-43f5-a813-ecd9222a1b46)

**Comment by bfabio** [comment]
> However I found a little bug in this PR where nav-sidebar is present on mobile... If you open the menu the sidebar is there on the bottom... but not really a bug, maybe a feature? 🤔

I'd say this is better UX-wise, that bottom-operated side bar is kinda weird anyway and keeping the bottom visible is a good visual hint that helps the user understanding where they are IMO.

File: src/components/header-menu-link/header-menu-link.scss (hunk 4 above)
Comment (bfabio) [inline src/components/header-menu-link/header-menu-link.scss:113]:
```suggestion
```

File: src/components/highlight/highlight.scss (hunk 3 above)
Comment (bfabio) [inline src/components/highlight/highlight.scss:25]:
```suggestion
```

File: src/components/nav-sidebar/nav-sidebar.scss [hunk 5]
```diff
@@ -86,7 +87,7 @@
 //Mobile & Tablet portrait only -------------------------------------------------------------------------
 @include media-breakpoint-down(lg) {
   .menu-column {
-    z-index: 10010 !important;
+    z-index: bootstrap.$zindex-modal;
```
Comment (Fupete) [inline src/components/nav-sidebar/nav-sidebar.scss:90]: Ummmmm così però su mobile non è sopra l'header come dovrebbe... Deve essere superiore a 10000 sennò rimane sotto il fixed (10000) e lo sticky (9999) 🤷 @bfabio

Comment (Fupete) [reply src/components/nav-sidebar/nav-sidebar.scss:90]: Questa sidebar non può appoggiarsi allo stack nativo di BSI finchè non rivediamo quello. È un caso molto particolare con intorno tanti livelli a cui essere sopra **solo** su mobile.

Comment (bfabio) [reply src/components/nav-sidebar/nav-sidebar.scss:90]: @Fupete AFAIU lo stacking dipende dal contesto di stacking, non è detto che siano numeri assoluti globali che si possono confrontare tra loro

Comment (Fupete) [reply src/components/nav-sidebar/nav-sidebar.scss:90]: Sì è che anche su BSI quei due numeri sono numeri...

**Comment by bfabio** [comment]
@Fupete wdyt?

**Review by bfabio — APPROVED** [review]

**Comment by Fupete** [comment]
Perfect for me. Let's merge this and restart from here. cc @astagi

**Pull request merged**